### PR TITLE
Fruit store hw

### PR DESCRIPTION
**Pull request by AlexPravda1**
- used Strategy pattern
- handler Map is done manually in main()
- checks against wrong initial data and negative fruit amount are implemented

### CODE REVIEW

File: src/main/java/dao/FileAccessDaoCsv.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package dao;
+
+import java.util.List;
+
+public interface FileAccessDaoCsv {
```
Comment (RomanDubovskyi) [inline src/main/java/dao/FileAccessDaoCsv.java:5]:
```suggestion
public interface CsvFileService {
```

File: src/main/java/dao/FileAccessDaoCsvImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            return Files.readAllLines(Path.of(filePath));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file located at: " + filePath);
+        }
+    }
+
+    @Override
+    public void writeToFile(String stringForReport, String outFilePath) {
+        File toFile = new File(outFilePath);
+        ifExists(toFile);
```
Comment (RomanDubovskyi) [inline src/main/java/dao/FileAccessDaoCsvImpl.java:25]: can we just use file.exists?

File: src/main/java/service/ActivityStrategyImpl.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    private final Map<String, ActivityHandler> activityHandlerMap;
+
+    public ActivityStrategyImpl(Map<String, ActivityHandler> activityHandlerMap) {
+        this.activityHandlerMap = activityHandlerMap;
+    }
+
+    @Override
+    public ActivityHandler get(String activityType) {
+
+        if (activityType.equals("b") || activityType.equals("p")
```
Comment (RomanDubovskyi) [inline src/main/java/service/ActivityStrategyImpl.java:16]: that violates O from SOLID, instead use: `if (!map.containsKey(activityCode)) { throw an exception }`

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/dao/FileAccessDaoCsvImpl.java (hunk 2 above)
Comment (AlexPravda1) [reply src/main/java/dao/FileAccessDaoCsvImpl.java:25]: thx, wasn't aware of the method until started "tests" HW. 
Removed this check for current task as non-relevant.

File: src/main/java/service/ActivityStrategyImpl.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    @Override
+    public ActivityHandler get(String activityType) {
+
+        if (activityHandlerMap.containsKey(activityType)) {
+            return activityHandlerMap.get(activityType);
+        } else {
+            throw new RuntimeException("Wrong activity type in the file");
+        }
```
Comment (RomanDubovskyi) [inline src/main/java/service/ActivityStrategyImpl.java:20]:
```suggestion
            return activityHandlerMap.get(activityType);
        } 
        throw new RuntimeException("Wrong activity type in the file");
```
we don't need else if we have return in the `if` block

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**